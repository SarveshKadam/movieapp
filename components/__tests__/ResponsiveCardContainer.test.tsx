import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ScrollView, View, Text } from 'react-native';
import ResponsiveCardContainer from '../CardContainer';

describe('ResponsiveCardContainer Component', () => {
  it('renders correctly with given children', () => {
    const tree = TestRenderer.create(
      <ResponsiveCardContainer>
        <View>
          <Text>Child 1</Text>
        </View>
        <View>
          <Text>Child 2</Text>
        </View>
      </ResponsiveCardContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('applies styles correctly', () => {
    const tree = TestRenderer.create(
      <ResponsiveCardContainer>
        <View>
          <Text>Child 1</Text>
        </View>
        <View>
          <Text>Child 2</Text>
        </View>
      </ResponsiveCardContainer>
    );
    const instance = tree.root;
    const scrollView = instance.findByType(ScrollView);

    expect(scrollView.props.contentContainerStyle).toMatchObject({
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'space-between',
      padding: 10,
    });
  });
});
