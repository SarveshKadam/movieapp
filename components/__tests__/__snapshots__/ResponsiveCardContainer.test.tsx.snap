// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponsiveCardContainer Component renders correctly with given children 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
>
  <View>
    <View>
      <Text>
        Child 1
      </Text>
    </View>
    <View>
      <Text>
        Child 2
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
